pkg/operator/controller/ingress/metrics_test.go

```diff
@@ -0,0 +1,126 @@
+				{
+					[]string{"default", "Available"},
+					1.0,
+				},
+				{
+					[]string{"default", "Degraded"},
+					0.0,
+				},
+				{
+					[]string{"test1", "Available"},
+					0.0,
+				},
+				{
+					[]string{"test1", "Degraded"},
+					1.0,
+				},
+			},
+			inputIngress: testIngressControllerWithConditions("test1", []operatorv1.OperatorCondition{
+				{Type: "Available", Status: operatorv1.ConditionFalse},
+				{Type: "Degraded", Status: operatorv1.ConditionTrue},
+			}),
+			expectedMetricFormat: `
+            # HELP ingress_controller_conditions Report the conditions for ingress controllers. 0 is False and 1 is True.
+            # TYPE ingress_controller_conditions gauge
+            ingress_controller_conditions{condition="Available",name="default"} 1
+            ingress_controller_conditions{condition="Degraded",name="default"} 0
+            `,
+		},
+		{
+			name: "Not reported conditions",
+			inputMetricValues: []metricValue{
+				{
+					[]string{"default", "Available"},
+					1.0,
+				},
+				{
+					[]string{"default", "Degraded"},
+					0.0,
+				},
+				{
+					[]string{"test1", "Available"},
+					0.0,
+				},
+				{
+					[]string{"test1", "Degraded"},
+					1.0,
+				},
+			},
+			inputIngress: testIngressControllerWithConditions("test1", []operatorv1.OperatorCondition{
+				{Type: "Available", Status: operatorv1.ConditionFalse},
+				{Type: "Degraded", Status: operatorv1.ConditionTrue},
+				{Type: "Admitted", Status: operatorv1.ConditionTrue},
+				{Type: "PodsScheduled", Status: operatorv1.ConditionTrue},
+			}),
+			expectedMetricFormat: `
+            # HELP ingress_controller_conditions Report the conditions for ingress controllers. 0 is False and 1 is True.
+            # TYPE ingress_controller_conditions gauge
+            ingress_controller_conditions{condition="Available",name="default"} 1
+            ingress_controller_conditions{condition="Degraded",name="default"} 0
+            `,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// cleanup the ingress condition metrics
+			ingressControllerConditions.Reset()
+
+			// fill the metric up with the input values
+			for _, val := range tc.inputMetricValues {
+				ingressControllerConditions.WithLabelValues(val.labels...).Set(val.value)
+			}
+
+			// check the testutil collected all the metrics
+			gotNumMetrics := testutil.CollectAndCount(ingressControllerConditions)
+			if gotNumMetrics != len(tc.inputMetricValues) {
+				t.Errorf("collected a diffent number of metrics before deletion. expected: %d, got: %d", len(tc.inputMetricValues), gotNumMetrics)
+			}
+
+			DeleteIngressControllerConditionsMetric(tc.inputIngress)
+
+			// check the remaning metrics
+			err := testutil.CollectAndCompare(ingressControllerConditions, strings.NewReader(tc.expectedMetricFormat))
+			if err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+type metricValue struct {
+	labels []string
+	value  float64
+}
```

Comment:
Moving this type definition closer to its use would improve readability (especially since the definitions that use this type don't use explicit field keys).

Reply:
Moved at the beginning of the file